### ISSUE TITLE
_cargo_build_script_run not executing relative to crate?

### ISSUE THREAD

**Issue by parisholley** [bazelbuild/rules_rust#391]
i am having issues with crates such as ring/openssl that attempt to compile libraries as part of the build... they try to refer to files that are relative to the crate (when vendor'd) but instead, the working directory is my repository root. if i manually edit the build files to do ./cargo/vendor/crate-name-version/path/to/file.c then it works fine (but obviously isn't a solution).

**Comment by damienmg** [comment]
Hi, sorry for the delay, I am a bit overloaded at the moment,

Would you mind sharing the Cargo.toml file you used so I can repro? Can you also share some details about your operating system?

**Comment by parisholley** [comment]
https://github.com/parisholley/cargobug

**Comment by damienmg** [comment]
Thanks for the repro. Sorry I still don't follow fully, what are you changing in the BUILD file?

**Comment by parisholley** [comment]
if you checkout the repo, and run the instructions in the readme, you can reproduce the error. i'm on Mac OS 10.15.5

**Comment by damienmg** [comment]
Yes I can repro, this was not what I wanted to ask. You said earlier that:

    if i manually edit the build files to do ./cargo/vendor/crate-name-version/path/to/file.c then it works fine (but obviously isn't a solution).

Are you editing the build.rs?

**Comment by parisholley** [comment]
yep

**Comment by UebelAndre** [comment]
Does this happen when using `remote` mode with `cargo raze`?

**Comment by dae** [comment]
@parisholley did #427 solve your problem?

**Comment by UebelAndre** [comment]
Closed by https://github.com/bazelbuild/rules_rust/pull/427
